gramex/config.py

```diff
@@ -840,3 +842,55 @@ def used_kwargs(method, kwargs, ignore_keywords=False):
             target = used if key in set(argspec.args) else rest
             target[key] = val
     return used, rest
+
+
+esl_paths = AttrDict()
+esl_paths['source'] = Path(__file__).absolute().parent
+esl_conf = (+PathConfig(esl_paths['source'] / 'gramex.yaml')).get('eslog', AttrDict())
+# The above config needs to be fixed. Gramex is loading this as a service
+allowed_log_levels = ["INFO","ERROR","DEBUG","WARN"]
+
+
+class ConnSingleton:
+    def __init__(self, cls):
+        self.cls = cls
+        self.instance = None
+
+    def __call__(self, *args, **kwargs):
+        if self.instance == None:
+            self.instance = self.cls(*args, **kwargs)
+        return self.instance
+
+
+@ConnSingleton
+class ElDb:
+    connection = None
+    def get_connection(self):
+        if self.connection is None or not self.connection.ping():
+            self.connection = Elasticsearch(esl_conf['host'],
+                http_auth=(esl_conf['user'],esl_conf['pass'])) 
+        return self.connection 
+
+
+def log(**kwargs):
+    '''
+    Writes the log into Elastic Search and application log.
+    Usage:
+        log(level='INFO',x=1, y=2, msg='log string') writes to ES as below
+        {'level': 'INFO', 'x': 1, 'y': 2, 'msg': 'log string', 'time': '2020-07-21 13:41:00', 
+            'port': 9988}
+    '''
+    log_args = kwargs.copy()
+    if 'level' in log_args.keys():
+        log_level = log_args['level'].upper()
+        if not log_level in allowed_log_levels:
+            log_args["level"] = "UNKNOWN"
+        log_args["level"] = log_level
```

Comment:
The above logic can be simplified. 

```python
log_level = log_args.get('level', 'UNKNOWN').upper()
log_args['level'] = allowed_log_levels.get(log_level, 'UNKNOWN')
```

Also, please use single quotes, not double-quotes